src/ui.rs

```diff
@@ -3687,41 +3687,38 @@ impl CodeWindow {
             return;
         }
         ui.should_redraw = true;
-        let mut breakpoint_id = None;
-        for (id, breakpoint) in debugger.breakpoints.iter() {
-            match &breakpoint.on {
-                BreakpointOn::Line(bp) if bp.path == tab.path_in_symbols && (bp.line == tab.area_state.cursor + 1 || bp.adjusted_line == Some(tab.area_state.cursor + 1)) => {
-                    if delete {
-                        debugger.remove_breakpoint(id);
-                        return;
-                    }
-                    breakpoint_id = Some(id.clone());
-                    if !edit_condition {
-                        let enable = !breakpoint.enabled;
-                        let r = debugger.set_breakpoint_enabled(id, enable);
-                        report_result(state, &r);
-                    }
-                    break;
-                }
-                _ => (),
-            }
-        }
+        let line_breakpoint = Self::make_breakpoint_for_current_line(tab);
+        let breakpoint_id = match line_breakpoint {
```

Comment:
Just change `make_breakpoint_for_current_line` to return a LineBreakpoint, and wrap it with BreakpointOn when calling add_breakpoint and step_to_cursor.